platform/openide.actions/src/org/openide/actions/HeapView.java

```diff
@@ -350,60 +148,39 @@ public HeapView() {
         setToolTipText(NbBundle.getMessage(GarbageCollectAction.class, "CTL_GC"));
         updateUI();
     }
-    
+
     /**
-     * Overriden to return true, GCComponent paints in its entire bounds in
-     * an opaque manner.
+     * Overridden to return true, GCComponent paints in its entire bounds in an
+     * opaque manner.
      */
-    @Override public boolean isOpaque() {
+    @Override
+    public boolean isOpaque() {
         return true;
     }
-    
+
     /**
      * Updates the look and feel for this component.
      */
-    @Override public void updateUI() {
-        Font f = new JLabel().getFont();
-        f = new Font(f.getName(), Font.BOLD, f.getSize());
+    @Override
+    public void updateUI() {
+        Font f = new JLabel().getFont().deriveFont(Font.BOLD);
```

Comment:
Probably better use `UIManager.getFont("Label.font")` instead of `new JLabel().getFont()`.

Reply:
Makes sense. While we're at it, is this the correct use of the updateUI method? The comments on `JComponent::updateUI` say:
```
     * Resets the UI property to a value from the current look and feel.
     * <code>JComponent</code> subclasses must override this method
     * like this:
     * <pre>
     *   public void updateUI() {
     *      setUI((SliderUI)UIManager.getUI(this);
     *   }
     *  </pre>
```
which I don't really understand.

Reply:
In this custom component it is OK. I think you could remove invocation of `revalidate()` and `repaint()` from `updateUI()`. Does not make any sense IMHO.

`JComponent::updateUI` is usually invoked from two places:
- the component constructor and
- when switching look and feel without restart, which is currently not supported by NB

`setUI()` and `UIManager.getUI()` are used if the component has a UI delegate, which is another class that implements painting, listeners, etc. E.g. UI delegate for `JLabel` is class `javax.swing.plaf.LabelUI`, which is sub-classed by LaFs.

`HeapView` does not have/need a UI delegate. You could even remove `HeapView.updateUI()` and initialize the font in the constructor. But maybe NB supports LaF switching in the future. Then it is good to have it because the label font may be different in various LaFs.